lighthouse-core/audits/webapp-install-banner.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+const Audit = require('./audit');
+const SWAudit = require('./service-worker');
+const Formatter = require('../formatters/formatter');
+
+/**
+ * @fileoverview
+ * Audits if a page is configured to prompt users with the webapp install banner.
+ * https://github.com/GoogleChrome/lighthouse/issues/23#issuecomment-270453303
+ *
+ * Requirements:
+ *   * manifest is not empty
+ *   * manifest has valid start url
+ *   * manifest has a valid name
+ *   * manifest has a valid shortname
+ *   * manifest display property is either standalone or fullscreen
+ *   * manifest contains icon that's a png and size >= 144px
+ *   * SW is registered, and it owns this page and the manifest's start url
+ *   * Site engagement score of 2 or higher
+
+ * This audit covers these requirements with the following exceptions:
+ *   * it doesn't look at standalone/fullscreen
```

Comment:
It seems wrong to not validate something which is in fact a real world requirement.

Maybe now would be a good time to resolve the question of whether Lighthouse is an expression of Chrome's policy, in which case I'm keen to continue to use this forum to advocate for the requirements to change to be in line with web standards, or if it is not, append "... in Google Chrome" to the end of the title of this audit (and I'll toddle off to file a bug in Chrome!).

I'm hopeful of convincing Samsung to drop the fullscreen/standalone requirement for an install prompt.